src/tds/bulk.c

```diff
@@ -760,16 +761,28 @@ tds7_bcp_send_colmetadata(TDSSOCKET *tds, TDSBCPINFO *bcpinfo)
 		 * different from BCP format
 		 */
 		if (is_blob_type(bcpcol->on_server.column_type)) {
-			/* FIXME support multibyte string */
-			len = tds_dstr_len(&bcpinfo->tablename);
-			TDS_PUT_SMALLINT(tds, len);
-			tds_put_string(tds, tds_dstr_cstr(&bcpinfo->tablename), len);
+			converted_name = tds_convert_string(tds, tds->conn->char_convs[client2ucs2], tds_dstr_cstr(&bcpinfo->tablename), (int)tds_dstr_len(&bcpinfo->tablename), &converted_len);
+			if (!converted_name) {
+				return TDS_FAIL;
```

Comment:
Should I `tds_set_state` hire prior to returning?

Reply:
As you probably will loose sync with the server is a good idea to disconnect.
I won't expect to happen, mostly would fail for memory errors.